### PR TITLE
encryption: few improvements to help debug and ginkgo

### PR DESCRIPTION
**Pull request by jrfastab**
I got ginkgo running on a self-managed cluster but had a couple issues resolved here,

- compile error on my bpf-next kernel and configurations.
- timers on cilium-health-ep are a bit tight and caused cilium-health agent to regenerate unnecessarily
- debugging default-route lookup with encryption error message was a bit mysterious make it better

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9383)

### CODE REVIEW

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
Recommending backport into 1.6 as these are simple patches and fix a compile error, plus get ginkgo running without flakes on my clusters at least.

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

File: pkg/datapath/linux/route/route.go [hunk 1]
```diff
@@ -88,12 +88,11 @@ func (r *Route) ToIPCommand(dev string) []string {
 func lookupDefaultRoute(family int) (netlink.Route, error) {
 	routes, err := netlink.RouteListFiltered(family, &netlink.Route{Dst: nil}, netlink.RT_FILTER_DST)
 	if err != nil {
-		log.WithError(err).Error("Unable to list direct routes")
```
Comment (joestringer) [inline pkg/datapath/linux/route/route.go:91]: Is this log message printed elsewhere or just not useful?

**Review by joestringer — APPROVED** [review]

File: pkg/datapath/linux/route/route.go (hunk 1 above)
Comment (jrfastab) [reply pkg/datapath/linux/route/route.go:91]: After patch its rolled up to daemon and printed in Fatal message above. I've also added in update to the other caller so now both call traces here will print this in the Fatal message.

**Comment by jrfastab** [comment]
test-me-please

File: daemon/health.go [hunk 2]
```diff
@@ -92,7 +92,7 @@ func (d *Daemon) initHealth() {
 				d.cleanupHealthEndpoint()
 				return err
 			},
-			RunInterval: 30 * time.Second,
+			RunInterval: 60 * time.Second,
```
Comment (brb) [inline daemon/health.go:95]: Just a thought, but if the higher interval is needed just during the bootstrap phase, perhaps we could set it back to 30s after the first successful run. Not sure whether it's possible with the current controller implementation though.

**Review by brb — APPROVED** [review]
LGTM, just one comment. Can be resolved in a follow-up.

**Comment by brb** [comment]
Needs rebasing.

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/15361/

```09:43:05      Check connectivity with managed etcd [It]
09:43:05      /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/k8s-1.16-gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:420
09:43:05  
09:43:05      Test Panicked
09:43:05      runtime error: index out of range
09:43:05      /usr/local/go/src/runtime/panic.go:44
09:43:05  
09:43:05      Full Stack Trace
09:43:05          /usr/local/go/src/runtime/panic.go:522 +0x1b5
09:43:05      github.com/cilium/cilium/test/k8sT.fetchPodsWithOffset(0xc0002016c0, 0x19dbec2, 0x6, 0x19e71ed, 0xd, 0xc0008dce68, 0xd, 0x2, 0xc000954c70, 0x10, ...)
09:43:05          /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/k8s-1.16-gopath/src/github.com/cilium/cilium/test/k8sT/DatapathConfiguration.go:332 +0xa49
09:43:05      github.com/cilium/cilium/test/k8sT.testPodConnectivityAcrossNodesAndReturnIP(0xc0002016c0, 0x1, 0x2, 0x1739ce0, 0x0)
09:43:05          /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/k8s-1.16-gopath/src/github.com/cilium/cilium/test/k8sT/DatapathConfiguration.go:350 +0x347
09:43:05      github.com/cilium/cilium/test/k8sT.testPodConnectivityAcrossNodes(...)
09:43:05          /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/k8s-1.16-gopath/src/github.com/cilium/cilium/test/k8sT/DatapathConfiguration.go:306
09:43:05      github.com/cilium/cilium/test/k8sT.glob..func2.14.2()
09:43:05          /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/k8s-1.16-gopath/src/github.com/cilium/cilium/test/k8sT/DatapathConfiguration.go:299 +0xa1
09:43:05      reflect.Value.call(0x1739ce0, 0xc000363940, 0x13, 0x19d8d31, 0x4, 0x2e0be20, 0x0, 0x0, 0xc000614120, 0x0, ...)
09:43:05          /usr/local/go/src/reflect/value.go:447 +0x461
09:43:05      reflect.Value.Call(0x1739ce0, 0xc000363940, 0x13, 0x2e0be20, 0x0, 0x0, 0x4460ac, 0x0, 0x19d26e0)
09:43:05          /usr/local/go/src/reflect/value.go:308 +0xa4
09:43:05      github.com/cilium/cilium/test/ginkgo-ext.applyAdvice.func1(0x2e0be20, 0x0, 0x0, 0x0, 0x0, 0x0)
09:43:05          /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/k8s-1.16-gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:467 +0x90
09:43:05      reflect.callReflect(0xc0004ce240, 0xc0008dd490, 0xc0008dd478)
09:43:05          /usr/local/go/src/reflect/value.go:536 +0x326
09:43:05      reflect.makeFuncStub(0xc000000020, 0x1aa7000, 0xc0008dd4c7, 0xc000384ba0, 0xc0008dd4e8, 0xc0008dd4e0, 0x400, 0xc0008dd668, 0x886c8f, 0xc000384ba0, ...)
09:43:05          /usr/local/go/src/reflect/asm_amd64.s:20 +0x42
09:43:05      github.com/cilium/cilium/vendor/github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc000384ba0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
09:43:05          /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/k8s-1.16-gopath/src/github.com/cilium/cilium/test/test_suite_test.go:101 +0x2f8
09:43:05      testing.tRunner(0xc00031cf00, 0x1aa5310)
09:43:05          /usr/local/go/src/testing/testing.go:865 +0xc0
09:43:05      created by testing.(*T).Run
09:43:05          /usr/local/go/src/testing/testing.go:916 +0x35a
09:43:05      
09:43:05  ------------------------------```

**Comment by jrfastab** [comment]
seems unrelated to my change going to try again, plus this was passing I simple rebased sha.

**Comment by jrfastab** [comment]
test-me-please

**Pull request merged**